test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -1212,6 +1212,14 @@ def _validate_argument_spec(self, docs, spec, kwargs, context=None):
                     code='parameter-alias-repeated',
                     msg=msg
                 )
+            data_choices = data.get('choices', set())
+            if not context and arg == 'state' and data_choices:
+                bad_states = {'list', 'info'} & set(data_choices)
+                for bad_state in bad_states:
+                    self.reporter.error(
+                        path=self.object_path,
+                        code='parameter-state-invalid-choice',
+                        msg="Argument 'state' includes the value '%s' as a valid choice" % bad_state)
```

Comment:
Typo?
```suggestion
                        msg="Argument 'state' includes the value '%s' as an invalid choice" % bad_state)
```

Reply:
I meant "it has been defined as 'valid'/'accepted' with the implication of an error being you shouldn't be doing this"

Reply:
Hopefully this is less confusing:
Argument 'state' includes the value '%s' as a choice

Reply:
Well, explicit is better than implicit so that it's not misleading ;)